### PR TITLE
improve(rds): feature polardb add data source

### PR DESCRIPTION
**Pull request by zhangyuangang**

### CODE REVIEW

File: alicloud/data_source_alicloud_polardb_accounts.go [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+			}
+			ids = append(ids, item.AccountName)
+			accounts = append(accounts, mapping)
+		}
+	}
+
+	d.SetId(dataResourceIdHash(ids))
+	if err := d.Set("accounts", accounts); err != nil {
+		return WrapError(err)
+	}
```
Comment (Lexsss) [inline alicloud/data_source_alicloud_polardb_accounts.go]: if err := d.Set("names", ids); err != nil {
		return WrapError(err)
	}

File: alicloud/data_source_alicloud_polardb_accounts.go [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+			"db_cluster_id": {
+				Type:     schema.TypeString,
+				ForceNew: true,
+				Required: true,
+			},
+			"name": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+
```
Comment (Lexsss) [inline alicloud/data_source_alicloud_polardb_accounts.go]: names

File: alicloud/data_source_alicloud_polardb_accounts_test.go [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+
+	var existPolarClusterMapFunc = func(rand int) map[string]string {
+		return map[string]string{
+			"accounts.#": CHECKSET,
+		}
+	}
+
+	var fakePolarClusterMapFunc = func(rand int) map[string]string {
+		return map[string]string{
+			"accounts.#": CHECKSET,
```
Comment (Lexsss) [inline alicloud/data_source_alicloud_polardb_accounts_test.go]: "accounts.#": 0,

File: alicloud/data_source_alicloud_polardb_accounts_test.go [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+			"db_cluster_id": `"${alicloud_polardb_cluster.default.id}"`,
+		}),
+		fakeConfig: testAccCheckAlicloudPolarClusterAccountsDataSourceConfig(rand, map[string]string{
+			"db_cluster_id": `"${alicloud_polardb_cluster.default.id}"`,
+		}),
+	}
+
+	var existPolarClusterMapFunc = func(rand int) map[string]string {
+		return map[string]string{
+			"accounts.#": CHECKSET,
```
Comment (Lexsss) [inline alicloud/data_source_alicloud_polardb_accounts_test.go]: check the attributes please

**Comment by zhangyuangang** [comment]
image

File: alicloud/data_source_alicloud_polardb_accounts_test.go [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+
+	dbClusterIdConf := dataSourceTestAccConfig{
+		existConfig: testAccCheckAlicloudPolarClusterAccountsDataSourceConfig(rand, map[string]string{
+			"db_cluster_id": `"${alicloud_polardb_cluster.default.id}"`,
+		}),
+		fakeConfig: testAccCheckAlicloudPolarClusterAccountsDataSourceConfig(rand, map[string]string{
+			"db_cluster_id": `"${alicloud_polardb_cluster.default.id}"`,
+		}),
+	}
+
```
Comment (Lexsss) [inline alicloud/data_source_alicloud_polardb_accounts_test.go]: nameConf

File: alicloud/data_source_alicloud_polardb_databases.go [hunk 6]
```diff
@@ -0,0 +1,128 @@
...
+)
+
+func dataSourceAlicloudPolarDBDatabases() *schema.Resource {
+	return &schema.Resource{
+		Read: dataSourceAlicloudPolarDBDatabasesRead,
+
+		Schema: map[string]*schema.Schema{
+			"db_cluster_id": {
+				Type:     schema.TypeString,
+				Optional: true,
```
Comment (Lexsss) [inline alicloud/data_source_alicloud_polardb_databases.go]: Required

File: alicloud/data_source_alicloud_polardb_databases_test.go [hunk 7]
```diff
@@ -0,0 +1,73 @@
+package alicloud
```
Comment (Lexsss) [inline alicloud/data_source_alicloud_polardb_databases_test.go]: still in this testcase

File: alicloud/data_source_alicloud_polardb_accounts.go (hunk 2 above)
Comment (zhangyuangang) [reply alicloud/data_source_alicloud_polardb_accounts.go]: > names

done

File: alicloud/data_source_alicloud_polardb_accounts.go (hunk 1 above)
Comment (zhangyuangang) [reply alicloud/data_source_alicloud_polardb_accounts.go]: > if err := d.Set("names", ids); err != nil {
> return WrapError(err)
> }

done

File: alicloud/data_source_alicloud_polardb_accounts.go [hunk 8]
```diff
@@ -0,0 +1,128 @@
...
+			accounts = append(accounts, mapping)
+		}
+	}
+
+	d.SetId(dataResourceIdHash(ids))
+	if err := d.Set("accounts", accounts); err != nil {
+		return WrapError(err)
+	}
+
+	if err := d.Set("names", strings.Join(ids, ",")); err != nil {
```
Comment (Lexsss) [inline alicloud/data_source_alicloud_polardb_accounts.go]: set a slice is ok

File: alicloud/data_source_alicloud_polardb_accounts.go [hunk 9]
```diff
@@ -0,0 +1,128 @@
...
+		Read: dataSourceAlicloudPolarDBAccountsRead,
+
+		Schema: map[string]*schema.Schema{
+			"db_cluster_id": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"names": {
+				Type:     schema.TypeString,
+				Optional: true,
```
Comment (Lexsss) [inline alicloud/data_source_alicloud_polardb_accounts.go]: no optional

File: alicloud/data_source_alicloud_polardb_accounts_test.go [hunk 10]
```diff
@@ -0,0 +1,93 @@
...
+	  db_node_class     = "polar.mysql.x4.large"
+	  vswitch_id        = "${alicloud_vswitch.default.id}"
+	  description       = "${var.name}"
+	}
+
+	resource "alicloud_polardb_account" "account" {
+	  db_cluster_id     = "${alicloud_polardb_cluster.default.id}"
+	  account_name      = "tftestnormal"
+	  account_password  = "Test12345"
+      account_description   = "${var.name}"
```
Comment (Lexsss) [inline alicloud/data_source_alicloud_polardb_accounts_test.go]: please fmt it

File: alicloud/data_source_alicloud_polardb_databases.go [hunk 11]
```diff
@@ -0,0 +1,133 @@
...
+		Read: dataSourceAlicloudPolarDBDatabasesRead,
+
+		Schema: map[string]*schema.Schema{
+			"db_cluster_id": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"names": {
+				Type:     schema.TypeString,
+				Optional: true,
```
Comment (Lexsss) [inline alicloud/data_source_alicloud_polardb_databases.go]: still

File: alicloud/data_source_alicloud_polardb_databases.go [hunk 12]
```diff
@@ -0,0 +1,133 @@
...
+			ids = append(ids, item.DBName)
+			databases = append(databases, mapping)
+		}
+	}
+
+	d.SetId(dataResourceIdHash(ids))
+	if err := d.Set("databases", databases); err != nil {
+		return WrapError(err)
+	}
+	if err := d.Set("names", strings.Join(ids, ",")); err != nil {
```
Comment (Lexsss) [inline alicloud/data_source_alicloud_polardb_databases.go]: still

File: alicloud/data_source_alicloud_polardb_databases_test.go [hunk 13]
```diff
@@ -0,0 +1,90 @@
...
+	}
+	
+	variable "name" {
+	  default = "tf-testAccPolarClusterConfig_%d"
+	}
+	
+	resource "alicloud_polardb_cluster" "default" {
+	  db_type           = "MySQL"
+	  db_version        = "8.0"
+      pay_type          = "PostPaid"
```
Comment (Lexsss) [inline alicloud/data_source_alicloud_polardb_databases_test.go]: still

File: website/docs/d/polardb_accounts.html.markdown [hunk 14]
```diff
@@ -0,0 +1,54 @@
...
+    value = "${data.alicloud_polardb_accounts.default.accounts[0].account_name}"
+}
+```
+
+## Argument Reference
+
+The following arguments are supported:
+
+* `db_cluster_id` - (Required) PolarDB cluster ID. 
+* `names` - (Optional) Account name of the cluster.
```
Comment (Lexsss) [inline website/docs/d/polardb_accounts.html.markdown]: no filter logic with `names`, so remove it in arguments please.

File: website/docs/d/polardb_databases.html.markdown [hunk 15]
```diff
@@ -0,0 +1,55 @@
...
+    value = "${data.alicloud_polardb_databases.default.databases[0].db_name}"
+}
+```
+
+## Argument Reference
+
+The following arguments are supported:
+
+* `db_cluster_id` - (Required) PolarDB cluster ID. 
+* `name` - (Optional) database name of the cluster.
```
Comment (Lexsss) [inline website/docs/d/polardb_databases.html.markdown]: still

File: alicloud/data_source_alicloud_polardb_databases.go [hunk 16]
```diff
@@ -0,0 +1,129 @@
...
+
+func dataSourceAlicloudPolarDBDatabases() *schema.Resource {
+	return &schema.Resource{
+		Read: dataSourceAlicloudPolarDBDatabasesRead,
+
+		Schema: map[string]*schema.Schema{
+			"db_cluster_id": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
```
Comment (xiaozhu36) [inline alicloud/data_source_alicloud_polardb_databases.go]: still applies

File: website/docs/d/polardb_accounts.html.markdown [hunk 17]
```diff
@@ -0,0 +1,53 @@
...
+layout: "alicloud"
+page_title: "Alicloud: alicloud_polardb_accounts"
+sidebar_current: "docs-alicloud-datasource-polardb-accounts"
+description: |-
+    Provides a collection of PolarDB endpoints according to the specified filters.
+---
+
+# alicloud\_polardb\_accounts
+
+The `alicloud_polardb_accounts` data source provides a collection of PolarDB account available in Alibaba Cloud account.
```
Comment (xiaozhu36) [inline website/docs/d/polardb_accounts.html.markdown]: I think the description is not correct. Please correct it.

File: website/docs/d/polardb_databases.html.markdown [hunk 18]
```diff
@@ -0,0 +1,54 @@
...
+layout: "alicloud"
+page_title: "Alicloud: alicloud_polardb_databases"
+sidebar_current: "docs-alicloud-datasource-polardb-databases"
+description: |-
+    Provides a collection of PolarDB endpoints according to the specified filters.
+---
+
+# alicloud\_polardb\_databases
+
+The `alicloud_polardb_databases` data source provides a collection of PolarDB database available in Alibaba Cloud account.
```
Comment (xiaozhu36) [inline website/docs/d/polardb_databases.html.markdown]: still applies

File: website/docs/d/polardb_databases.html.markdown [hunk 19]
```diff
@@ -0,0 +1,54 @@
...
+output "ends" {
+    value = "${data.alicloud_polardb_databases.default.databases[0].db_name}"
+}
+```
+
+## Argument Reference
+
+The following arguments are supported:
+
+* `db_cluster_id` - (Required) PolarDB cluster ID.
```
Comment (xiaozhu36) [inline website/docs/d/polardb_databases.html.markdown]: please keep sync with schema

**Pull request merged**